articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,172 @@
+
+
+Now that you have a broad idea of what we are about to build, let's jump right in!
+
+## Aim
+1. Deploy two applications and have it managed by NGINX.
+
+## Setup & Pre-Requisites:
+
+1. For this example, I have two sample Express Applications. One can have any kind of application running on different ports. 
+**NOTE:** Do not run your application on Port 80 or 443. It will be explained later why this must not be done.  
+Refer the official ExpressJS documentation for Getting Started. [Docs](https://expressjs.com/en/starter/installing.html)
+2. I have installed NGINX on my local machine, but the same could be done on any Virtual Machine where the applications are expected to be deployed.  
+Here is the documentation on how to install NGINX on your machine. [Docs](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/) . 
+
+## Step 1: Start two apps running in different ports
+As I've mentioned earlier, I've got two Node.js Apps running on two different ports as shown below.  
+
+**Server app running on Port 3000**
+
+![image](./server1.png)  
+
+**Client app running on Port 3001**  
+
+![image](./client1.png)
+
+Now that we have our apps up and running, we don't want our users to use these applications by typing their PORTS explicitly, so we need to map it with something which is more human-readable. In this example, I will be using subdomains to distinguish between them. Again one is free to use whichever distinguishing element suitable as per their requirement.
+Another example could be a particular route like domain/client and domain/server. The only condition for the distinguishing element is to follow a Valid URL Regular Expression. Learn about Regex from [here](https://regexr.com/).
+
+
+## Step 2: Add DNS records
+This is the part where one would add the DNS records in their DNS management dashboard. If you are running Nginx locally, you can skip this step.
+
+The general DNS Configurations would be something like
+- Server app mapped to the server.domain
+- Client app mapped to the client.domain
+
+My Localhost Config, in this case, would be
+- Server mapped to server.localhost
+- Client mapped to client.localhost
+
+There are two standard protocols HTTP and HTTPS. The default port for HTTP is 80 and HTTPS is 443. This is the reason we must not run our applications on these ports because our NGINX server would be running on these two ports. All the requests the client makes would either be redirected to Port 80 or 443 from where it would be redirected internally to the corresponding application.
+
+## Step 4 - Configure NGINX at 80 for HTTP and 443 for HTTPS
+Now that we have our apps running and our DNS records ready. We can start configuring our NGINX Reverse Proxy to make it all work.  
+By default, the configuration file is named nginx.conf and placed in the directory /usr/local/nginx/conf, /etc/nginx, or /usr/local/etc/nginx for Linux and Debian Based systems.
```

Comment:
On my Debian install of Nginx, the `nginx.conf` was in /etc/nginx.

Reply:
Great that is where you are expected to see the nginx.conf file.

Also I took a look at your conf file which you have on PasteBin. There is an extra closing parenthesis on line 70, which closes the http block. so if you could remove line 70, you should be good to go. I tried testing the corrected conf file on a debian machine, and the test was successful. Let me know if that solves the problem.

Reply:
Thanks, @HarishTeens for your help. Removed Line 70 but got `nginx: [emerg] "server" directive is not allowed here in /etc/nginx/nginx.conf:71` for Line 71.

Reply:
I am terribly sorry, your old config was right with the parenthesis on the 71st line. Just figured out that you need to add a ```;``` to lines 86,87,96,97 . I have updated the code snippet in the article as well.